src/build-roc.ts

```diff
@@ -115,8 +115,10 @@ const buildRocFile = (
   const errors = []
   const buildingForMac = target.startsWith("macos") || (target === "" && os.platform() === "darwin")
   const buildingForLinux = target.startsWith("linux") || (target === "" && os.platform() === "linux")
-  const rocBuildOutputDir = os.tmpdir()
-  const rocBuildOutputFile = path.join(rocBuildOutputDir, rocFileName.replace(/\.roc$/, ".o"))
+  const tmpDir = os.tmpdir()
+  const rocBuildOutputDir = fs.mkdtempSync(`${tmpDir}${path.sep}`)
+  const targetSuffix = (target === "" ? "native" : target)
+  const rocBuildOutputFile = path.join(rocBuildOutputDir, rocFileName.replace(/\.roc$/, `-${targetSuffix}.o`))
```

Comment:
What about making `rocFileName` unique (it's always `main` so it can still collide if the architecture is the same in concurrent builds)

Reply:
Oh, sorry, I missed random temp directory creation in the middle. disregard my previous comment